src/ui/components/search/searchcomponent.js

```diff
@@ -348,26 +376,46 @@ export default class SearchComponent extends Component {
     if (this.iconState === iconState) {
       return;
     }
+    this.prevState = this.iconState;
     this.iconState = iconState;
     if (!this.isRequestingAnimationFrame) {
       this.isRequestingAnimationFrame = true;
       window.requestAnimationFrame(() => {
-        this.forwardIcon.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
-        this.reverseIcon.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
+        DOM.query(this._container, this.prevState).classList.add('yxt-SearchBar-Icon--inactive');
+        const iconEl = DOM.query(this._container, this.iconState);
+        iconEl.classList.remove('yxt-SearchBar-Icon--inactive');
+        iconEl.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
         if (this.iconState === IconState.MAGNIFYING_GLASS) {
-          this.forwardIcon.classList.remove('yxt-SearchBar-AnimatedIcon--inactive');
-          this.reverseIcon.classList.add('yxt-SearchBar-AnimatedIcon--inactive');
-        } else if (this.iconState === IconState.YEXT) {
-          this.forwardIcon.classList.add('yxt-SearchBar-AnimatedIcon--inactive');
-          this.reverseIcon.classList.remove('yxt-SearchBar-AnimatedIcon--inactive');
+          const forwardEl = DOM.query(this._container, '.Icon--yext_animated_forward');
+          this.prevState === IconState.LOADING || this.prevState === IconState.CUSTOM_LOADING
+            ? forwardEl.classList.add('yxt-SearchBar-MagnifyingGlass')
```

Comment:
Ok I just thought of this but would adding a `--static` suffix make things a little clearer? Maybe we add a comment here saying why we need this, since it's kind of a specific, nonsensical (from a code perspective) product request?

Reply:
hmm sounds good, will do!